### ISSUE TITLE
[telemetry] CLIENT spans for SFTP operations in ph-oton-connect

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#51]
## Goal

Emit ph-telemetry `CLIENT` spans for SFTP operations in `ph-oton-connect`, and turn the existing session statistics into an up-down counter of open sessions.

These are the only genuine **outbound network calls** in ph-oton. A `CLIENT` span here is what makes a remote-server hang show up as a gap in the trace instead of as an unexplained slow request.

## Where

**1. `ph-oton-connect/src/main/java/com/helger/photon/connect/sftp/ChannelSftpRunner.java`**

```java
public static ESuccess execute (@NonNull IJSchSessionProvider aSessionProvider,
                                int nChannelConnectTimeoutMillis,
                                @NonNull IChannelSftpRunnable aRunnable) throws JSchException

public static ESuccess execute (@NonNull IJSchSessionProvider aSessionProvider,
                                @Nullable Duration aChannelConnectTimeout,
                                @NonNull IChannelSftpRunnable aRunnable) throws JSchException
```

The first overload just delegates to the second — instrument only the second, so a call is not double-counted.

It declares exactly **one** checked exception, so `Telemetry.<ESuccess, JSchException> withSpanThrowing (...)` applies cleanly. This is the case the throwing variants were designed for; no signature change is needed.

**2. `ph-oton-connect/src/main/java/com/helger/photon/connect/sftp/JSchSessionFactory.java`**

Already has two `StatisticsManager` counters, `STATS_COUNTER_CREATED` and `STATS_COUNTER_DESTROYED`, incremented in `createSession (...)` and `destroySession (...)`. **Keep them** and add an up-down counter alongside, so the number of *currently open* sessions becomes visible — the difference between the two existing counters is exactly what is interesting and exactly what a monitoring system cannot compute from them today.

## What to emit

New `com.helger.photon.connect.CConnectTelemetry`:

| Constant | Value |
|---|---|
| `SPAN_SFTP_EXECUTE` | `photon.sftp.execute` |
| `METRIC_SFTP_OPERATIONS` | `photon.sftp.operations` |
| `METRIC_SFTP_DURATION` | `photon.sftp.duration` |
| `METRIC_SFTP_SESSIONS_OPEN` | `photon.sftp.sessions.open` |
| `ATTR_SFTP_HOST` | `photon.sftp.host` |
| `ATTR_SFTP_PORT` | `photon.sftp.port` |
| `ATTR_SFTP_OPERATION` | `photon.sftp.operation` (from `IChannelSftpRunnable`, see below) |
| `ATTR_SFTP_SUCCESS` | `photon.sftp.success` |
| `UNIT_OPERATION` | `{operation}` |
| `UNIT_SESSION` | `{session}` |
| `UNIT_MILLIS` | `ms` |

- **Span** `photon.sftp.execute`, kind **`CLIENT`** (not `INTERNAL` — this is an outbound call), attributes host, port and operation name.
- **Counter** `photon.sftp.operations`, attributes host + success.
- **Histogram** `photon.sftp.duration` in ms, attributes host + success.
- **Up-down counter** `photon.sftp.sessions.open` — `+1` next to `STATS_COUNTER_CREATED.increment ()`, `-1` next to `STATS_COUNTER_DESTROYED.increment ()`, with identical attributes on both sides so it nets out to zero.

Check what `IChannelSftpRunnable` exposes for a name/description before wiring `ATTR_SFTP_OPERATION`; if there is no bounded name available, drop that attribute rather than using the runnable's `toString ()`.

## Cardinality

- Host and port are bounded per deployment (a handful of configured servers) and are fine as metric attributes.
- Do **not** put file names, remote paths, or user names on metrics — they are unbounded. File paths may go on the **span** if useful, and that is where `SftpConnector` / `SftpConnectionDestination` detail belongs.
- Credentials from `IServerConnectionSettingsPassword` / `IServerConnectionSettingsKeyPair` must never appear anywhere in telemetry.

## POM

`ph-oton-connect/pom.xml` needs a new dependency (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

`ph-oton-connect` is a leaf module that no other ph-oton module depends on, so this addition is fully contained.

## Acceptance criteria

- [ ] Only the second `execute (...)` overload is instrumented; the delegating overload adds no second span.
- [ ] Span kind is `CLIENT`.
- [ ] Existing `StatisticsManager` counters untouched; up-down counter balanced.
- [ ] No credentials, remote paths or user names on metric attributes.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.6.0**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-connect/.../connect/CConnectTelemetry.java` | new, public - span / metric / attribute names |
| `ph-oton-connect/.../connect/sftp/SftpMetrics.java` | new, public - counter, histogram, up-down counter |
| `ph-oton-connect/.../connect/sftp/SftpTelemetry.java` | new, package-private emitter |
| `ph-oton-connect/.../connect/sftp/ChannelSftpRunner.java` | modified |
| `ph-oton-connect/.../connect/sftp/JSchSessionFactory.java` | modified - 3 call sites |
| `ph-oton-connect/pom.xml` | `com.helger.telemetry:ph-telemetry` added (no `<version>`, BOM) |

### Only the second overload

The deprecated `execute (IJSchSessionProvider, int, IChannelSftpRunnable)` just delegates and was left completely untouched, so a call through it produces exactly one span.

Its body moved into a new private `_executeInSpan (...)` and the public method now reads:

```java
return Telemetry.<ESuccess, JSchException> withSpanThrowing (CConnectTelemetry.SPAN_SFTP_EXECUTE,
                                                             ETelemetrySpanKind.CLIENT,
                                                             aSpan -> { ... });
```

`JSchException` is the single checked exception the method declares, so the throwing variant fits without any signature change - this is the case those helpers were designed for. Extracting the body was necessary to keep the lambda readable; the logic inside is unchanged.

**Span kind is `CLIENT`**, as required - these are the only genuine outbound network calls in ph-oton.

### Emission

* Span `photon.sftp.execute` with `photon.sftp.operation` (the runnable's display name), plus `photon.sftp.host` and `photon.sftp.port` as soon as the session is established.
* Counter `photon.sftp.operations` and histogram `photon.sftp.duration`, both with `host` + `success`, emitted from a `finally` around the span so that a failure to even create the session is counted too. The host is carried out of the lambda in a `Wrapper <String>`; if the session was never created, the attribute is simply absent (`TelemetryAttributes.Builder.put` drops a `null` value).
* Up-down counter `photon.sftp.sessions.open`: `+1` next to each of the two `STATS_COUNTER_CREATED.increment ()` calls, `-1` next to `STATS_COUNTER_DESTROYED.increment ()`. Both sides build their attributes through the same private `_getSessionAttrs (Session)`, so they are identical by construction and the counter nets out to zero. Both existing `StatisticsManager` counters are untouched.

The runner catches `SftpException` and returns `ESuccess.FAILURE` without re-throwing, so the span helper never sees it - `SftpTelemetry.onOperationError (...)` records the exception and sets the status explicitly.

### Cardinality

`IChannelSftpRunnable extends IHasDisplayName`, and the display name exists for error messages - in practice it embeds the remote file name, which is unbounded. It is therefore a **span attribute only**, not a metric attribute; the issue's fallback ("drop that attribute rather than using `toString ()`") is honoured in spirit by keeping it off the metrics entirely. Host and port come from `Session.getHost ()` / `getPort ()` and are bounded per deployment. Credentials from `IServerConnectionSettingsPassword` / `IServerConnectionSettingsKeyPair` never appear anywhere.

### Verification

Full reactor `mvn install` over all 23 modules: BUILD SUCCESS, no test failures. No unit test was added - exercising this path needs a real SFTP server, and the issue's acceptance criteria did not ask for one. `README.md` news bullet added under `v10.6.0 - work in progress`.
